docs/architecture/adr-064-abci-2.0.md

```diff
@@ -347,9 +338,14 @@ legacy ABCI types, e.g. `LegacyBeginBlockRequest` and `LegacyEndBlockRequest`. O
 we can come up with new types and names altogether.
 
 ```go
-func (app *BaseApp) FinalizeBlock(req abci.RequestFinalizeBlock) abci.ResponseFinalizeBlock {
-	// merge any state changes from ProcessProposal into the FinalizeBlock state
-	app.MergeProcessProposalState()
+func (app *BaseApp) FinalizeBlock(req abci.RequestFinalizeBlock) (*abci.ResponseFinalizeBlock, error) {
+	ctx := ...
+
+	if app.preFinalizeBlockHook != nil {
```

Comment:
Does this mean that if the user defines a vote extension they need to define this hook?

Reply:
Not necessarily, no. Only where you need some state derived from VE to be stored.